elessar/pages/Home/home.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { Component} from 'react';
+import { View, FlatList, ActivityIndicator, ScrollView, StyleSheet, Text} from 'react-native';
+import { getUsers} from '../Login/requests'
+
+
+interface HomeState {
+    users: any,
+    loading: boolean,
+    limit: number,
+    page: number,
+    offset: number,
+    finished: boolean,
+}
+export default class Home extends Component<{}, HomeState>{
+    constructor(props) {
+        super(props);
+        this.state = {
+            users: [],
+            loading: false,
+            limit: 20,
+            page: 1,
+            offset: 0,
+            finished: false,
+        }
+    }
+
+
+    loadUsers = async () => {
+        if (this.state.loading) return;
+
+        if(!this.state.finished){
+            const { page } = this.state;
+            this.setState({ loading: true });
+            const response = (await getUsers(this.state.limit*(this.state.page-1), this.state.limit))
+            const usersToConcat = response.data?.users?.nodes;
+            this.setState({finished: !response.data?.users?.pageInfo?.hasNextPage})
+            this.setState({
+                users: [ ...this.state.users, ...usersToConcat ],
+                page: page + 1,
+            });   
+        }
+        this.setState({ loading: false });
+
+    
+        
+      }
+
+    componentDidMount = async () =>{
+        this.loadUsers();        
+    }
```

Comment:
Quando usamos normalmente (sem precisar do `this`) seria possível manter a sintaxe de função mesmo, que é mais simples e intuitiva.

```ts 
async componentDidMount() {
  this.loadUsers();
}
```